xayn-ai/src/ranker/mod.rs

```diff
@@ -237,8 +234,10 @@ mod tests {
 
     #[test]
     fn test_rank_no_documents() {
+        let mut documents: Vec<TestDocument> = vec![];
+
         let res = rank(
-            &mut [],
+            &mut documents,
```

Comment:
we could add the type hint directly in the call, then the construction in l.237 wouldn't be needed.
```suggestion
            &mut [] as &mut [TestDocument],
```